scripts/clang-format.sh

```diff
@@ -0,0 +1,591 @@
+#!/bin/bash
+#
```

Comment:
I was bashing around. Sorry. Kinda became a complete script.

The goal of this script is 

- to be the backing of the make targets
- provide reformatting of all commits on a branch functionality
- provide nicer error messages if something is not nicely set up, e.g. missing install of clang-format
- isolate the ubuntu clang versionitis, e.g "oh, I installed clang-format-10, so I should call git clang-format-10, not git clang-format..."

Best reviewed from the bottom up, i.e. start at the end.